Core/Core/CourseSync/CourseSyncSelector/Model/CourseSyncSelectorInteractor.swift

```diff
@@ -0,0 +1,291 @@
+                        id: course.courseId,
+                        tabs: mappedTabs,
+                        files: files
+                    )
+                }
+                .eraseToAnyPublisher()
+        } else {
+            return Just(
+                CourseSyncSelectorEntry(
+                    name: course.name,
+                    id: course.courseId,
+                    tabs: mappedTabs,
+                    files: []
+                )
+            )
+            .setFailureType(to: Error.self)
+            .eraseToAnyPublisher()
+        }
+    }
+
+    private func getAllFiles(courseId: String) -> AnyPublisher<[CourseSyncSelectorEntry.File], Error> {
+        unowned let unownedSelf = self
+
+        return ReactiveStore(
+            useCase: GetFolderByPath(
+                context: .course(courseId)
+            )
+        )
+        .getEntities()
+        .flatMap {
+            Publishers.Sequence(sequence: $0)
+                .filter { !$0.lockedForUser && !$0.hiddenForUser }
+                .setFailureType(to: Error.self)
+                .flatMap { unownedSelf.getFiles(folderID: $0.id, initialArray: []) }
+        }
+        .map {
+            $0.map {
+                CourseSyncSelectorEntry.File(
+                    id: $0.file?.id ?? Foundation.UUID().uuidString,
+                    name: $0.file?.displayName ?? "Unknown file",
+                    url: $0.file?.url
+                )
+            }
+        }
+        .replaceEmpty(with: [])
+        .eraseToAnyPublisher()
+    }
+
+    private func getFiles(folderID: String, initialArray: [FolderItem]) -> AnyPublisher<[FolderItem], Error> {
+        unowned let unownedSelf = self
+
+        var result = initialArray
+
+        return getFilesAndFolderIDs(folderID: folderID)
+            .flatMap { files, folderIDs in
+                result.append(contentsOf: files)
+
+                guard folderIDs.count > 0 else {
+                    return Just([result])
+                        .setFailureType(to: Error.self)
+                        .eraseToAnyPublisher()
+                }
+                return Publishers.Sequence(sequence: folderIDs)
+                    .setFailureType(to: Error.self)
+                    .flatMap {
+                        unownedSelf.getFiles(
+                            folderID: $0,
+                            initialArray: result
+                        )
+                        .handleEvents(receiveOutput: { result = $0 })
+                    }
+                    .collect()
+                    .eraseToAnyPublisher()
+            }
+            .first()
+            .map { _ in result }
+            .eraseToAnyPublisher()
+    }
+
+    private func getFilesAndFolderIDs(folderID: String) -> AnyPublisher<([FolderItem], [String]), Error> {
+        ReactiveStore(
+            useCase: GetFolderItems(
+                folderID: folderID
+            )
+        )
+        .getEntities()
+        .retry(3)
```

Comment:
As I remember we eliminated all those http 403 calls, do we still need this?

Reply:
Yes, we probably don't need this anymore. Removed it.